### PR TITLE
Issue #1201 Converted test_nocasedict.py from unittest to pytest and changed test_function decorator

### PR DESCRIPTION
**Pull request by andy-maier**
For details, see the commit messages (two commits).
Ready for review and merge.

### CODE REVIEW

File: pywbem/_nocasedict.py [hunk 1]
```diff
@@ -368,19 +368,28 @@ def __repr__(self):
 
     def update(self, *args, **kwargs):
         """
-        Update the dictionary from sequences of key/value pairs provided in any
-        positional arguments, and from key/value pairs provided in any keyword
-        arguments. The key/value pairs are not copied.
+        Update the dictionary from key/value pairs. If an item for a key
+        exists in the dictionary, its value is updated. If an item for a key
+        does not exist, it is added to the dictionary at the end.
+        The provided keys and values are not copied.
```
Comment (KSchopmeyer) [inline pywbem/_nocasedict.py]: This sentence takes some thinking.  We mean that the values from the dictionary are used without copying them.

File: pywbem/_nocasedict.py [hunk 2]
```diff
@@ -368,19 +368,28 @@ def __repr__(self):
...
 
         Each positional argument can be:
 
           * an object with a method `items()` that returns an
             :term:`py:iterable` of tuples containing key and value.
+            A NocaseDict or other dict object would match that requirement.
```
Comment (KSchopmeyer) [inline pywbem/_nocasedict.py]: Don't understand this sentence

File: pywbem/_nocasedict.py [hunk 3]
```diff
@@ -368,19 +368,28 @@ def __repr__(self):
...
 
           * an object without such a method, that is an :term:`py:iterable` of
             tuples containing key and value.
+            A list of tuples (key,value) would match that requirement.
```
Comment (KSchopmeyer) [inline pywbem/_nocasedict.py]: Same here.  Not sure what you are trying to say.

File: pywbem/_nocasedict.py (hunk 1 above)
Comment (andy-maier) [reply pywbem/_nocasedict.py]: DONE.
Changed the sentence to: "The provided keys and values are stored in the dictionary without being copied."

File: pywbem/_nocasedict.py (hunk 2 above)
Comment (andy-maier) [reply pywbem/_nocasedict.py]: I wanted to give a hint what this kind of positional argument is used for, but I guess it is not necessary. Removed the sentence, and also the similar sentence in the next item.
DONE.

File: pywbem/_nocasedict.py (hunk 3 above)
Comment (andy-maier) [reply pywbem/_nocasedict.py]: See previous comment.

**Comment by andy-maier** [comment]
The new commit addresses all review comments.

**Pull request merged**